configs/vico/README.md

```diff
@@ -0,0 +1,148 @@
+
+<!-- [IMAGE] -->
+
+<div align=center>
+<img src="https://github.com/haoosz/ViCo/assets/71176040/0ee95a57-fecf-4bba-bc64-eda46e5cc6d1">
+</div>
+
+## Configs
+
+|       Model       | Dataset | Download |
+| :---------------: | :-----: | :------: |
+| [ViCo](./vico.py) |    -    |    -     |
+
+## Quick Start
+
+1. Download [data](https://drive.google.com/drive/folders/1m8TCsY-C1tIOflHtWnFzTbw2C6dq67mC) and [templates](https://drive.google.com/drive/folders/1SpByLKECISmj5fhkaicT4yrsyqqpWL_T)
+   and save to `data/vico/`
+
+The file structure will be like this:
+
+```text
+data
+└── vico
+    └──batman
+       ├── 1.jpg
+       ├── 2.jpg
+       ├── 3.jpg
+       └── 4.jpg
+    └──clock
+       ├── 1.jpg
+       ├── 2.jpg
+       ├── 3.jpg
+       └── 4.jpg
+    ...
+    └──imagenet_templates_small.txt
+```
+
+2. Customize your config
+
+```
+# Only need to care about these
+
+# which concept you want to customize
+concept_dir = 'dog7'
+
+# the new token to denote the concept
+placeholder: str = 'S*'
+
+# better to be the superclass of concept
+initialize_token: str = 'dog'
+```
+
+3. Start training with the following command:
+
+```bash
+# 4 GPUS
+bash tools/dist_train.sh configs/vico/vico.py 4
+# 1 GPU
+python tools/train.py configs/vico/vico.py
+```
+
+4. Use the [pretrained checkpoins](https://drive.google.com/drive/folders/1GQGVzzOP2IgEfsQ-6ii6o2DqElnFThHM) to inference
+
+```python
+import torch
+from mmengine import Config
+from PIL import Image
+
+from mmagic.registry import MODELS
+from mmagic.utils import register_all_modules
+
+register_all_modules()
+
+# say you have downloaded the pretrained weights
+cfg = Config.fromfile('configs/vico/dog.py')
+state_dict = torch.load("./dog.pth")
+vico = MODELS.build(cfg.model)
+vico.load_state_dict(state_dict)
+vico = vico.cuda()
+
+prompt = ["A photo of S*", "A photo of S* on the beach"]
+reference = "data/vico/dog7/01.jpg"
+image_ref = Image.open(reference)
+with torch.no_grad():
+    output = vico.infer(prompt=prompt, image_reference=image_ref, seed=123, num_images_per_prompt=2)['samples'][0]
+output.save("infer.png")
+```
+
+5. (Optional) If you want to use the weight trained by the
+   commands at step3, here are codes to extract the trained parameters, then you can infer with it like step4
+
+```python
+def extract_vico_parameters(state_dict):
+    new_state_dict = dict()
+    for k, v in state_dict.items():
+        if 'image_cross_attention' in k or 'trainable_embeddings' in k:
+            new_k = k.replace('module.', '')
+            new_state_dict[new_k] = v
+    return new_state_dict
+
+checkpoint = torch.load("work_dirs/vico/iter_400.pth")
+new_checkpoint = extract_vico_parameters(checkpoint['state_dict'])
+torch.save(new_checkpoint, "work_dirs/vico/dog.pth")
+```
+
```

Comment:
I trainded a toy model by following step3. After I have extracted ckpt like step5, I run infer like step4 and meet this error
![image](https://github.com/open-mmlab/mmagic/assets/12782558/a12c15c4-5dad-476b-82e7-17cf7af9872d)

Reply:
It's because light-weight version of ckpt only contains part of the params. So `strict=False` is needed when using `torch.load`. [Codes](https://github.com/open-mmlab/mmagic/pull/1978/files#diff-40d41ed519c7e44c2f8d680b7713273c027fb56d06a89109e7fa5e16cd298420R91) in README have been modified.